ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -1188,11 +1189,16 @@ class PartCentredForceHolder {
             plugin_.ReportGroundCollision(
                 closest_physical_parent(part1).flightID);
           }
-#if KSP_VERSION_1_9_1
-          foreach (var collider in part1.currentCollisions.Keys) {
-#elif KSP_VERSION_1_7_3
-          foreach (var collider in part1.currentCollisions) {
-#endif
+          var collision_reporter =
+              part1.gameObject.GetComponent<CollisionReporter>();
```

Comment:
Is this going to be expensive?  I think we had performance problems with this kind of operation in the past.

Reply:
We are getting a component of this specific game object, and it should almost always be there.
We do kind of thing every frame for the navball already:
https://github.com/mockingbirdnest/Principia/blob/aef945b1f8c1def933d3d8f74494abfefa6e1085/ksp_plugin_adapter/ksp_plugin_adapter.cs#L1725-L1731 https://github.com/mockingbirdnest/Principia/blob/aef945b1f8c1def933d3d8f74494abfefa6e1085/ksp_plugin_adapter/ksp_plugin_adapter.cs#L1836-L1847.


I think you are thinking of incidents involving [`FindObjectOfType`](https://github.com/mockingbirdnest/Principia/pull/1489/files), which looks at all objects; that is slow.